internal/file/group_test.go

```diff
@@ -91,3 +91,59 @@ func TestCheckFilePathDependantCasesWithLockFile(t *testing.T) {
 	check = group.checkFilePathDependantCases(false, false, "file.txt")
 	assert.True(t, check)
 }
+
+func TestGroupMatchLockFile(t *testing.T) {
+	var match bool
+	g1 := NewGroup("requirements.txt", nil, []string{})
+	match = g1.matchLockFile("requirements.txt.pip.debricked.lock", "")
+	assert.Equal(t, match, true)
+	g2 := NewGroup("/home/requirements-test.txt", nil, []string{})
+	match = g2.matchLockFile("requirements-test.txt.pip.debricked.lock", "/home/")
+	assert.Equal(t, match, true)
+	g3 := NewGroup("requirements.txt", nil, []string{})
+	match = g3.matchLockFile("requirements.dev.txt.pip.debricked.lock", "")
+	assert.Equal(t, match, false)
+	g4 := NewGroup("requirements-test.txt", nil, []string{})
+	match = g4.matchLockFile("requirements.txt.pip.debricked.lock", "")
+	assert.Equal(t, match, false)
+	g5 := NewGroup("requirements-test.txt", nil, []string{})
+	match = g5.matchLockFile("requirements.txt-test.txt.pip.debricked.lock", "")
+	assert.Equal(t, match, false)
+
+	// Check that match fails if different directories
+	g6 := NewGroup("requirements.txt", nil, []string{})
+	match = g6.matchLockFile("requirements.txt.pip.debricked.lock", "some/other/directory")
+	assert.Equal(t, match, false)
+
+	// Check that match fails if there is no manifest file - functionality may change in the future.
+	g7 := NewGroup("", nil, []string{})
+	match = g7.matchLockFile("requirements.txt.pip.debricked.lock", "some/other/directory")
+	assert.Equal(t, match, false)
+}
+
+func TestGroupMatchManifestFile(t *testing.T) {
+	var match bool
+	g1 := NewGroup("", nil, []string{"/home/requirements.txt.pip.debricked.lock"})
+	match = g1.matchManifestFile("requirements.txt", "/home/")
+	assert.Equal(t, match, true)
+
+	// Check that match fails if different directories
+	g6 := NewGroup("", nil, []string{"requirements.txt.pip.debricked.lock"})
+	match = g6.matchManifestFile("requirements.txt", "some/other/directory")
+	assert.Equal(t, match, false)
+
+	// Check that match fails when manifest file exists -- only one per group for now.
+	g7 := NewGroup("requirements.txt", nil, []string{})
+	match = g7.matchManifestFile("requirements.dev.txt", "")
+	assert.Equal(t, match, false)
+}
+
+func TestGroupMatchFile(t *testing.T) {
+	var match bool
+	match = matchFile("/home/package-test.json", "/home/package-test-lock.json")
+	assert.Equal(t, match, true)
+	match = matchFile("/home/package.json", "/home/package-test-lock.json")
+	assert.Equal(t, match, false)
+	match = matchFile("/home/package-test.json", "/home/package-lock.json")
+	assert.Equal(t, match, false)
```

Comment:
Where can I read up on this special case of npm? :blush:

Reply:
Great question, I went back to look for the source, but I can't seem to find it anymore. Considering the time spent looking, I think whichever source I read was wrong. I'm removing this functionality.